packages/cli/create-strapi-app/src/utils/template.ts

```diff
@@ -0,0 +1,212 @@
+    const filePath = template.startsWith('file://') ? url.fileURLToPath(template) : template;
+
+    await fse.copy(filePath, rootPath);
+  }
+
+  if (isGithubShorthand(template)) {
+    const [owner, repo, ...pathSegments] = template.split('/');
+    const subPath = pathSegments.length ? pathSegments.join('/') : scope.templatePath;
+
+    await retry(
+      () => downloadGithubRepo(rootPath, { owner, repo, branch: scope.templateBranch, subPath }),
+      {
+        retries: 3,
+        onRetry(err, attempt) {
+          console.log(`Retrying to download the template. Attempt ${attempt}. Error: ${err}`);
+        },
+      }
+    );
+
+    return;
+  }
+
+  if (isGithubRepo(template)) {
+    const url = new URL(template);
+
+    const [owner, repo, t, branch, ...pathSegments] = stripTrailingSlash(
+      url.pathname.slice(1)
+    ).split('/');
+
+    if (t !== undefined && t !== 'tree') {
+      throw new Error(`Invalid GitHub template URL: ${template}`);
+    }
+
+    if (scope.templateBranch) {
+      await downloadGithubRepo(rootPath, {
+        owner,
+        repo,
+        branch: scope.templateBranch,
+        subPath: scope.templatePath,
+      });
+    }
+
+    await retry(
+      () =>
+        downloadGithubRepo(rootPath, {
+          owner,
+          repo,
+          branch: decodeURIComponent(branch) ?? scope.templateBranch,
+          subPath: pathSegments.length
+            ? decodeURIComponent(pathSegments.join('/'))
+            : scope.templatePath,
+        }),
+      {
+        retries: 3,
+        onRetry(err, attempt) {
+          console.log(`Retrying to download the template. Attempt ${attempt}. Error: ${err}`);
+        },
+      }
+    );
+
+    throw new Error(`Invalid GitHub template URL: ${template}`);
+  }
+}
+
+type RepoInfo = {
+  owner: string;
+  repo: string;
+  branch?: string;
+  subPath?: string | null;
+};
+
+async function downloadGithubRepo(rootPath: string, { owner, repo, branch, subPath }: RepoInfo) {
+  const filePath = subPath ? subPath.split('/').join(path.posix.sep) : null;
+
+  let checkContentUrl = `https://api.github.com/repos/${owner}/${repo}/contents`;
+  if (filePath) {
+    checkContentUrl = `${checkContentUrl}/${filePath}`;
+  }
+
+  if (branch) {
+    checkContentUrl = `${checkContentUrl}?ref=${branch}`;
+  }
+
+  const checkRes = await fetch(checkContentUrl, {
+    method: 'HEAD',
+  });
+
+  if (checkRes.status !== 200) {
+    throw new Error(
+      `Could not find a template at https://github.com/${owner}/${repo}${branch ? ` on branch ${branch}` : ''}${filePath ? ` at path ${filePath}` : ''}`
+    );
+  }
+
+  let url = `https://api.github.com/repos/${owner}/${repo}/tarball`;
+
+  if (branch) {
+    url = `${url}/${branch}`;
+  }
+
+  const res = await fetch(url);
+
+  if (!res.body) {
+    throw new Error(`Failed to download ${url}`);
+  }
+
+  await pipeline(
+    // @ts-expect-error - Readable is not a valid source
+    Readable.fromWeb(res.body),
```

Comment:
Can't you directly create a compatible readable stream like this? (haven't tested if it actually work, but the types should be compatible)

```suggestion
    new ReadableStream(res.body),
```

Reply:
nah it didn't work when I tried and I found this method in the next.js cli I believe